airflow/utils/dag_processing.py

```diff
@@ -22,17 +22,39 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+import logging
+import multiprocessing
 import os
 import re
+import signal
+import sys
 import time
 import zipfile
 from abc import ABCMeta, abstractmethod
 from collections import defaultdict
+from datetime import timedelta
 
+import psutil
+from sqlalchemy import or_
+from tabulate import tabulate
+
+# To avoid circular imports
+import airflow.models
+from airflow import configuration as conf
 from airflow.dag.base_dag import BaseDag, BaseDagBag
 from airflow.exceptions import AirflowException
 from airflow.utils import timezone
+from airflow.utils.db import provide_session
 from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.state import State
+
+python_version_info = sys.version_info
+if python_version_info.major > 2:
```

Comment:
maybe a comment here, and how we do handle this kinda version dependent change at other places in airflow?

Reply:
Definitely, I'll put comment in next iteration. We have a couple of places also branching based on `sys.version_info`( mostly because of the `str` to `unicode` mess).

Reply:
saw this change https://github.com/apache/incubator-airflow/pull/3298 and will try to use that more decent approach